### DISCUSSION TITLE
Hook object return type reference stability

### DISCUSSION THREAD

**Discussion by ryanberckmans** [wevm/wagmi#4101] [General]
```typescript
const r = useEstimateGas(a);
useEffect(() => console.log("this re-fires after every render because r is unconditionally a new object each render"), [r]);
```

Would it be possible for every wagmi hook to have reference stability across renders if no fields change?

I understand that many clients are satisfied to destructure the result and that options like `notifyOnChangeProps` can help reduce unnecessary rerenders, but I still think this may be useful.

**Comment by tmm** [comment]
If TanStack Query supports it, Wagmi supports it.
